# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/windowwatcher/nsWindowWatcher.cpp
# Commit: de69e3c1635a
# Full Hash: de69e3c1635a311b5b22716b65b93fc1899a37a4
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-08-01 09:48:10
# Regressor Bug: 1613431
# File Overlap Count: 1
# Description:
#   Bug 1613431 - Part 3: Handle synced setters return value. r=nika
#   
#   Depends on D83646
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83647
# ==============================================================================

diff -r b326d33f98cb -r de69e3c1635a toolkit/components/windowwatcher/nsWindowWatcher.cpp
--- a/toolkit/components/windowwatcher/nsWindowWatcher.cpp	Fri Jul 31 13:37:00 2020 +0000
+++ b/toolkit/components/windowwatcher/nsWindowWatcher.cpp	Fri Jul 31 13:37:13 2020 +0000
@@ -967,7 +967,7 @@
   // If our parent is sandboxed, set it as the one permitted sandboxed navigator
   // on the new window we're opening.
   if (activeDocsSandboxFlags && parentBC) {
-    newBC->SetOnePermittedSandboxedNavigator(parentBC);
+    MOZ_ALWAYS_SUCCEEDS(newBC->SetOnePermittedSandboxedNavigator(parentBC));
   }
 
   if (!aForceNoOpener && parentBC) {
@@ -998,7 +998,7 @@
   if (activeDocsSandboxFlags &
       SANDBOX_PROPAGATES_TO_AUXILIARY_BROWSING_CONTEXTS) {
     MOZ_ASSERT(windowIsNew, "Should only get here for new windows");
-    newBC->SetSandboxFlags(activeDocsSandboxFlags);
+    MOZ_ALWAYS_SUCCEEDS(newBC->SetSandboxFlags(activeDocsSandboxFlags));
   }
 
   RefPtr<nsGlobalWindowOuter> win(
@@ -1043,9 +1043,9 @@
      is not a window name. */
   if (windowNeedsName) {
     if (nameSpecified && !name.LowerCaseEqualsLiteral("_blank")) {
-      newBC->SetName(name);
+      MOZ_ALWAYS_SUCCEEDS(newBC->SetName(name));
     } else {
-      newBC->SetName(EmptyString());
+      MOZ_ALWAYS_SUCCEEDS(newBC->SetName(EmptyString()));
     }
   }
 
@@ -1123,7 +1123,7 @@
         // Make sure we don't mess up our counter even if the above assert
         // fails.
         if (!newBC->GetIsPopupSpam()) {
-          newBC->SetIsPopupSpam(true);
+          MOZ_ALWAYS_SUCCEEDS(newBC->SetIsPopupSpam(true));
         }
       }
     }
